src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashPostBuildComment.java

```diff
@@ -64,8 +65,13 @@ public DescriptorImpl() {
     }
 
     @Override
-    public StashPostBuildComment newInstance(StaplerRequest req, JSONObject formData)
+    public StashPostBuildComment newInstance(@Nullable StaplerRequest req, JSONObject formData)
         throws FormException {
+      if (req == null) {
```

Comment:
Can we use https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String- here?

Reply:
Let's see what others do.

In Jenkins, many calls don't use `req` or don't check it. `core/src/main/java/hudson/tasks/Maven.java` and `test/src/test/java/hudson/util/RobustReflectionConverterTest.java` throw `FormException`.

Some plugins that I checked use `assert req != null`. I'm not a fan of `assert`, as it needs to be explicitly enabled on the Java command line, and that would enable _all_ asserts.

I'm using SpotBugs with Eclipse, SpotBugs is a FindBugs replacement. The problem with `requireNonNull` doesn't resolve the warning, it actually creates a higher ranked warning with higher confidence.

SpotBugs warning for the original code:

> Bug: req must be non-null but is marked as nullable
> This parameter is always used in a way that requires it to be non-null, but the parameter is explicitly annotated as being Nullable. Either the use of the parameter or the annotation is wrong.
> 
> Rank: Of Concern (19), confidence: Low
> Pattern: NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE 
> Type: NP, Category: STYLE (Dodgy code)

SpotBugs warning for `requireNonNull`

> Bug: req must be non-null but is marked as nullable
> This parameter is always used in a way that requires it to be non-null, but the parameter is explicitly annotated as being Nullable. Either the use of the parameter or the annotation is wrong.
> 
> Rank: Of Concern (16), confidence: Normal
> Pattern: NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE 
> Type: NP, Category: STYLE (Dodgy code)

Of course, we don't have to throw `FormException`, we can throw any other exception. I still prefer `FormException`, as we know that the caller would be prepared to handle it.

Reply:
Have you tried doing it like this?
```
requireNonNull(req,"req").bindJSON(StashPostBuildComment.class, formData);
```

Reply:
I just tried, it didn't help.
```
Bug: req must be non-null but is marked as nullable
This parameter is always used in a way that requires it to be non-null, but the parameter is explicitly annotated as being Nullable. Either the use of the parameter or the annotation is wrong.

Rank: Of Concern (16), confidence: Normal
Pattern: NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE 
Type: NP, Category: STYLE (Dodgy code)
```